src/lib.rs

```diff
@@ -47,28 +69,71 @@ where
     strfmt_map(fmtstr, &formatter)
 }
 
+pub fn strfmt_ext<'a, K, T: TypeFormatting>(fmtstr: &str, vars: &HashMap<K, T>) -> Result<String>
+    where
+        K: Hash + Eq + FromStr
+{
+
+    let formatter = |mut fmt: Formatter| {
+        let k: K = match fmt.key.parse() {
+            Ok(k) => k,
+            Err(_) => {
+                return Err(new_key_error(fmt.key));
+            }
+        };
+        let v = match vars.get(&k) {
+            Some(v) => v,
+            None => {
+                return Err(new_key_error(fmt.key));
+            }
+        };
+
+        v.do_format(&mut fmt)
+    };
+    strfmt_map(fmtstr, &formatter)
+}
+
 pub trait Format {
```

Comment:
Sorry, I'll make this as clear as I can explain with my limited English. 
The marked trait ist defining functions for `str` and `String` where usage would be `"{Zero}".format(&vars)`.
By adding a legacy function here named `strfmt_display` it would make sense to add a similar extension to allow a usage like `"{Zero}".format_display(&vars)` as either the original signature would change from  `format<K, D: fmt::Display>(&self, vars: &HashMap<K, D>) -> Result<String>` to `format<K, D: DisplayStr>(&self, vars: &HashMap<K, D>) -> Result<String>` or the body would change and default to use the legacy code.

Don't hesitate to ask, sometimes I think way to complicated and by reading these lines I wrote I'm not sure if it gets clear what problem ist the cause for this..

Reply:
Thanks for clarifying. I agree with you, there should be two functioned defined by `Format`:


```
trait Format {
  fn format<K, D: DisplayStr>(&self, vars: &HashMap<K, D>) -> Result<String>

  fn format_display<K, D: Display>(&self, vars: &HashMap<K, D>) -> Result<String>
```